resource/utilities/rq2.cpp

```diff
@@ -69,16 +225,149 @@ int info (resource_query_t &ctx)
+                break;
+            case 'S': /* --match-subsystems */
+                json_object_set_new(options, "matcher_name", json_string(optarg));
+                break;
+            case 'P': /* --match-policy */
+                json_object_set_new(options, "matcher_policy", json_string(optarg));
+                break;
+            case 'F': /* --match-format */
+                json_object_set_new(options, "match_format", json_string(optarg));
+                if (!known_match_format (optarg)) {
+                    std::cerr << "[ERROR] unknown format for --match-format: ";
+                    std::cerr << optarg << std::endl;
+                    usage (1);
+                }
+                break;
+            // case 'g': /* --graph-format */
+            //     rc = string_to_graph_format (optarg, ctx->params.o_format);
+            //     if ( rc != 0) {
+            //         std::cerr << "[ERROR] unknown format for --graph-format: ";
+            //         std::cerr << optarg << std::endl;
+            //         usage (1);
+            //     }
+            //     graph_format_to_ext (ctx->params.o_format, ctx->params.o_fext);
+            //     break;
+            // case 'o': /* --graph-output */
+            //     ctx->params.o_fname = optarg;
+            //     break;
+            case 'p': /* --prune-filters */
+                token = optarg;
+                if(token.find_first_not_of(' ') != std::string::npos) {
+                    if (!json_object_get(options,"prune_filters")){
+                        json_object_set_new(options, "prune_filters", json_string(token.c_str()));
+                    }else{
+                      json_object_set_new(options, "prune_filters",
+                                          json_pack("s++",
+                                          json_object_get(options,"prune_filters"),
+                                           ",", token.c_str()));
+                    }
+
+                }
+                break;
+            // case 't': /* --test-output */
+            //     ctx->params.r_fname = optarg;
+            //     break;
+            case 'r': /* --reserve-vtx-vec */
+                // If atoi fails, it defaults to 0, which is fine for us
+                json_object_set_new(options, "match_format", json_integer(atoi(optarg)));
+                if ( (atoi(optarg) < 0)
+                    || (atoi(optarg) > 2000000)) {
+                    json_object_set_new(options, "match_format", json_integer(0));
+                    std::cerr
+                        << "WARN: out of range specified for --reserve-vtx-vec: ";
+                    std::cerr << optarg << std::endl;
+                }
+                break;
+            // case 'e': /* --elapse-time */
+            //     ctx->params.elapse_time = true;
+            //     break;
+            // case 'd': /* --disable-prompt */
+            //     ctx->params.disable_prompt = true;
+            //     break;
```

Comment:
As long as these are unhandled better to remove the code or throw an error and put a `// TODO` comment I think. But why are they unhandled? Is there some complication involved in getting them to work?

Reply:
Right now, we could handle them easily in this module. The main issue would be that the new resource query class doesn't use them in the constructor. I could go either way on how to deal with these, I could parse them here and put a TODO on the constructor, remove them for now, or leave as is.